packages/remix/src/utils/instrumentServer.ts

```diff
@@ -302,6 +309,46 @@ function getTraceAndBaggage(): {
   return {};
 }
 
+async function injectTraceAndBaggageToLoaderData(loaderData: AppData, remixVersion: number): Promise<AppData> {
+  const traceAndBaggage = getTraceAndBaggage();
+
+  if (isDeferredData(loaderData)) {
+    loaderData.data['sentryTrace'] = traceAndBaggage.sentryTrace;
+    loaderData.data['sentryBaggage'] = traceAndBaggage.sentryBaggage;
+    loaderData.data['remixVersion'] = remixVersion;
+
+    return loaderData;
+  }
+
+  if (isResponse(loaderData)) {
+    // Note: `redirect` and `catch` responses do not have bodies to extract.
+    // We skip injection of trace and baggage in those cases.
+    // For `redirect`, a valid internal redirection target will have the trace and baggage injected.
+    if (isRedirectResponse(loaderData) || isCatchResponse(loaderData)) {
+      DEBUG_BUILD && logger.warn('Skipping injection of trace and baggage as the response does not have a body');
+      return loaderData;
+    } else {
+      const data = await extractData(loaderData);
+
+      if (typeof data === 'object') {
```

Comment:
`typeof null` returns `object`

```suggestion
      if (data != null && typeof data === 'object') {
```